src/service-worker.js

```diff
@@ -31,52 +31,53 @@ self.addEventListener('activate', event => {
 	);
 });
 
+
+async function fetchAndCache(request) {
+	const cache = await caches.open(`offline${timestamp}`)
+
+	try {
+		const response = await fetch(request);
+		cache.put(request, response.clone());
+		return response;
+	} catch (err) {
+		const response = await cache.match(request);
+		if (response) return response;
+
+		throw err;
+	}
+}
+
 self.addEventListener('fetch', event => {
 	if (event.request.method !== 'GET' || event.request.headers.has('range')) return;
 
 	const url = new URL(event.request.url);
 
 	// don't try to handle e.g. data: URIs
-	if (!url.protocol.startsWith('http')) return;
+	const isHttp = url.protocol.startsWith('http');
+	const isDevServerRequest = url.hostname === self.location.hostname && url.port !== self.location.port;
+	const isStaticAsset = url.host === self.location.host && cached.has(url.pathname);
```

Comment:
you could move this down to line 65 where it's used

Reply:
`isStaticAsset`? It's also used on the next line (59).

Reply:
oh, I missed that. nevermind then :smile: